core/src/main/java/org/bitcoinj/uri/BitcoinURI.java

```diff
@@ -188,12 +188,15 @@ private BitcoinURI(String input, @Nonnull Network network) throws BitcoinURIPars
         }
 
         // Attempt to parse the rest of the URI parameters.
-        parseParameters(network, nameValuePairTokens);
+        parseParameters(nameValuePairTokens);
 
         if (!addressToken.isEmpty()) {
             // Attempt to parse the addressToken as a Bitcoin address for this network
             try {
-                Address address = new DefaultAddressParser().parseAddress(addressToken, network);
+                DefaultAddressParser addressParser = new DefaultAddressParser();
+                Address address = network != null ?
+                        addressParser.parseAddress(addressToken, network) :
+                        addressParser.parseAddressAnyNetwork(addressToken);
```

Comment:
This code should be able to be simplified in the future once Issue #3021 and Issue #3022 are implemented.